packages/application-generic/src/services/bull-mq.service.ts

```diff
@@ -64,8 +78,15 @@ export class BullmqService {
     const WorkerClass = !BullmqService.pro
       ? Worker
       : require('@taskforcesh/bullmq-pro').WorkerPro;
+
+    let internalOptions: WorkerOptions;
+    if (options) {
+      internalOptions = options;
+    }
+    internalOptions.metrics = { maxDataPoints: MetricsTime.ONE_MONTH };
```

Comment:
```suggestion
    let internalOptions: WorkerOptions = {};
    if (options) {
      internalOptions = options;
    }
    internalOptions.metrics = { maxDataPoints: MetricsTime.ONE_MONTH };
```

internalOptions.metrics might fail because internal options can be potentially undefined.

Reply:
```suggestion
    const internalOptions: WorkerOptions = {
       ...options,
       metrics: { maxDataPoints: MetricsTime.ONE_MONTH },
    }
```